spec/models/submission_spec.rb

```diff
@@ -18,4 +18,44 @@
     expect(FactoryGirl.build(:submission, age: -30)).not_to be_valid
     expect(FactoryGirl.build(:submission, age: 130)).not_to be_valid
   end
+
+  describe '#generate_confirmation_token!' do
+    let(:submission) { FactoryGirl.create(:submission) }
+
+    it 'generates confirmation token' do
+      submission.generate_confirmation_token!
+      expect(submission.confirmation_token).not_to be_nil
+      expect(submission.confirmation_token_created_at).not_to be_nil
+    end
+  end
+
+  describe "#has_expired?" do
+    it "returns true when token expired and confirmation status hasn't yet changed" do
+      submission = FactoryGirl.build(
+        :submission,
+        confirmation_status: 'awaiting',
+        confirmation_token_created_at: 1.week.ago - 1)
+      expect(submission.has_expired?).to be true
+    end
+
+    it "returns false when token expired but status is not awaiting" do
+      confirmed_submission = FactoryGirl.build(
+        :submission,
+        confirmation_status: 'confirmed',
+        confirmation_token_created_at: 1.week.ago - 1)
+      expired_submission = FactoryGirl.build(
+        :submission,
+        confirmation_status: 'expired',
+        confirmation_token_created_at: 1.week.ago - 1)
+      not_invited_submission = FactoryGirl.build(
+        :submission,
+        confirmation_status: 'not_avaible',
+        confirmation_token_created_at: 1.week.ago - 1)
+      submissions = [confirmed_submission, expired_submission, not_invited_submission]
+
+      expect(confirmed_submission).not_to be_has_expired
```

Comment:
This reads bad, I would probably write `expect(confirmed_submission.has_expired?).to be_false`